### PR TITLE
Add make uppercase/lowercase

### PR DESCRIPTION
**Pull request by scholtzan**
This adds functionality to make text uppercase and lowercase.
This relies on https://github.com/google/xi-editor/pull/523

### CODE REVIEW

File: XiEditor/EditViewController.swift [hunk 1]
```diff
@@ -259,6 +259,14 @@ class EditViewController: NSViewController, EditViewDataSource, FindDelegate, Sc
         editView.inputContext?.discardMarkedText()
         document.sendRpcAsync("select_all", params: [])
     }
+    
+    override func uppercaseWord(_ sender: Any?) {
+        document.sendRpcAsync("uppercase", params: [])
```
Comment (cmyr) [inline XiEditor/EditViewController.swift]: The way we handle system selectors is that we actually catch them in the `EditView` itself, in the `doCommand` method; there's a map there that translates from Cocoa commands to rpcs. As is neither of these will currently get called.

**Comment by scholtzan** [comment]
Thanks! I updated it. Before I could only make the text uppercase/lowercase by clicking on the menu entries. Now it also works using custom key bindings.

**Comment by cmyr** [comment]
Ah interesting, I didn't even know those menu items were there, and didn't know that they worked as implemented; hence my baroque workaround. Cool.

File: XiEditor/EditViewController.swift (hunk 1 above)
Comment (cmyr) [reply XiEditor/EditViewController.swift]: This is... ambiguous. The `Events` enum was a drive-by addition, and hasn't really been used consistently. I think the longer-term goal would be to have a `XiCore` protocol that is implemented by `CoreConnection` (or a thin wrapper) which exposes the API. In any case I don't mind this as is, I don't love the `Events` API particularly, and this is clear.

File: XiEditor/EditView.swift [hunk 2]
```diff
@@ -389,6 +389,8 @@ class EditView: NSView, NSTextInputClient, TextPlaneDelegate {
         "transpose:": "transpose",
         "yank:": "yank",
         "cancelOperation:": "cancel_operation",
+        "uppercaseWord": "uppercase",
```
Comment (raphlinus) [inline XiEditor/EditView.swift]: These need to have trailing colons (confirm that it works when changed, doesn't work for me as-is).

**Comment by cmyr** [comment]
@scholtzan something weird happened to this PR, you've dragged in some stale changes, not totally sure what's up.

**Comment by scholtzan** [comment]
Ooh, right! I was syncing my fork a couple of days ago to get the newest updates. 
I will try to fix it now.

**Comment by scholtzan** [comment]
Fixed. And lesson learned to do my changes on a separate branch. ;)

**Comment by raphlinus** [comment]
Thanks!

**Pull request merged**